src/app/planner/majors/Majors.tsx

```diff
@@ -0,0 +1,106 @@
+"use client";
+
+import { Requirements } from "@/app/components/modals/majors/Requirements";
+import UserProgramsEditor from "@/app/components/modals/majors/majorSelection/MajorSelection";
+import {
+  MajorVerificationContext,
+  MajorVerificationProvider,
+} from "@/app/contexts/MajorVerificationProvider";
+import { PlannerProvider } from "@/app/contexts/PlannerProvider";
+import {
+  PlannersContext,
+  PlannersProvider,
+} from "@/app/contexts/PlannersProvider";
+import { useUserPermissions, useUserPrograms } from "@/app/hooks/reactQuery";
+import { Program } from "@/app/types/Program";
+import {
+  extractUnexpiredPrograms,
+  hasPermissionToEditProgram,
+} from "@/lib/permissionsUtils";
+import { Card } from "@mui/joy";
+import { useSession } from "next-auth/react";
+import { usePathname, useRouter } from "next/navigation";
+import { useContext, useMemo } from "react";
+
+export default function Majors() {
+  return (
+    <PlannersProvider>
+      <MajorVerificationProvider>
+        <MidComponent />
+      </MajorVerificationProvider>
+    </PlannersProvider>
+  );
+}
+
+function MidComponent() {
```

Comment:
Not a very good name. Indicative that maybe we can inline this component inside of `Majors`

Reply:
I renamed it to SelectionContainer. I don't think we can inline this, I have to keep it in its own component in order to access the active planner from the context.